### PR TITLE
Parallelize kickmaps

### PR DESCRIPTION
**Pull request by Gabrielrezende-asc**
This PR adds WLS data and, most importantly, parallelizes kick map calculations.

### CODE REVIEW

File: idanalysis/kickmaps.py [hunk 1]
```diff
@@ -269,26 +284,52 @@ def fmap_calc_kickmap(self, posx, posy, beam_energy=None, rk_s_step=None):
...
+        else:
+            for i, ryi in enumerate(self.posy):
+                for j, rxi in enumerate(self.posx):
+                    self._config.traj_init_rx = 1e3 * rxi
+                    self._config.traj_init_ry = 1e3 * ryi
+                    IDKickMap._fmap_calc_traj(self._config)
+                    pxf = self._config.traj.px[-1]
+                    pyf = self._config.traj.py[-1]
+                    rxf = self._config.traj.rx[-1]
+                    ryf = self._config.traj.ry[-1]
```
Comment (fernandohds564) [inline idanalysis/kickmaps.py]: I think you can reuse the method `_calc kick_map_mp` here.

File: idanalysis/kickmaps.py [hunk 2]
```diff
@@ -253,7 +254,21 @@ def generate_linear_kickmap(
                 self.kickx[i, j] = pxf * brho**2
                 self.kicky[i, j] = pyf * brho**2
 
-    def fmap_calc_kickmap(self, posx, posy, beam_energy=None, rk_s_step=None):
+    def _calc_kickmap_mp(self, args):
```
Comment (fernandohds564) [inline idanalysis/kickmaps.py]: This method can be a staticmethod.

File: idanalysis/kickmaps.py [hunk 3]
```diff
@@ -269,26 +284,52 @@ def fmap_calc_kickmap(self, posx, posy, beam_energy=None, rk_s_step=None):
...
+                        rxi * 1e3, ryi * 1e3
+                    )
+                    stg += "px = {:.01f} urad, py = {:.01f} urad".format(
+                        pxf * 1e6, pyf * 1e6
+                    )
+                    print(stg)
+                    self.kickx[i, j] = pxf * brho**2
+                    self.kicky[i, j] = pyf * brho**2
+                    self.fposx[i, j] = rxf / 1e3
+                    self.fposy[i, j] = ryf / 1e3
```
Comment (fernandohds564) [inline idanalysis/kickmaps.py]: This part of the code could also be turned into a method, so you can reuse it in both types of calculation: parallel and serial.

**Review by murilobalves — APPROVED** [review]

File: idanalysis/kickmaps.py (hunk 1 above)
Comment (Gabrielrezende-asc) [reply idanalysis/kickmaps.py]: Solved!

File: idanalysis/kickmaps.py (hunk 2 above)
Comment (Gabrielrezende-asc) [reply idanalysis/kickmaps.py]: Solved!

File: idanalysis/kickmaps.py (hunk 3 above)
Comment (Gabrielrezende-asc) [reply idanalysis/kickmaps.py]: Solved!

**Review by vellosok75 — APPROVED** [review]

**Review by fernandohds564 — APPROVED** [review]

**Review by xresende — APPROVED** [review]

**Pull request merged**